frontend/src/lib/lemon-ui/LemonSwitch/LemonSwitch.tsx

```diff
@@ -44,6 +48,41 @@ export function LemonSwitch({
         conditionalProps['aria-label'] = ariaLabel
     }
 
+    let tooltipContent: JSX.Element | null = null
+    if (disabledReason) {
+        disabled = true // Support `disabledReason` while maintaining compatibility with `disabled`
+        tooltipContent = <span className="italic">{disabledReason}</span>
+    }
+    let buttonComponent = (
+        <button
+            id={id}
+            className="LemonSwitch__button"
+            role="switch"
+            onClick={() => {
+                if (onChange) {
+                    onChange(!checked)
+                }
+            }}
+            onMouseDown={() => setIsActive(true)}
+            onMouseUp={() => setIsActive(false)}
+            onMouseOut={() => setIsActive(false)}
+            data-attr={dataAttr}
+            disabled={disabled}
+            {...conditionalProps}
+        >
+            <div className="LemonSwitch__slider" />
+            <div className="LemonSwitch__handle">{handleContent}</div>
+        </button>
+    )
+    if (tooltipContent) {
+        buttonComponent = (
+            <Tooltip title={tooltipContent}>
+                {/* If the button is disabled, wrap it in a div so that the tooltip works */}
+                {disabled ? <div>{buttonComponent}</div> : buttonComponent}
```

Comment:
Does it work to always wrap it in a div?

Reply:
I guess it must....